src/main/java/frc/team2412/robot/commands/launcher/FullTargetCommand.java

```diff
@@ -0,0 +1,97 @@
+package frc.team2412.robot.commands.launcher;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.interpolation.InterpolatingTreeMap;
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import frc.team2412.robot.Controls;
+import frc.team2412.robot.Robot;
+import frc.team2412.robot.subsystems.DrivebaseSubsystem;
+import frc.team2412.robot.subsystems.IntakeSubsystem;
+import frc.team2412.robot.subsystems.LauncherSubsystem;
+import frc.team2412.robot.util.LauncherDataLoader;
+import frc.team2412.robot.util.LauncherDataPoint;
+import java.nio.file.FileSystems;
+import java.util.function.BooleanSupplier;
+
+public class FullTargetCommand extends Command {
+
+	private static final InterpolatingTreeMap<Double, LauncherDataPoint> LAUNCHER_DATA =
+			LauncherDataLoader.fromCSV(
+					FileSystems.getDefault()
+							.getPath(Filesystem.getDeployDirectory().getPath(), "launcher_data.csv"));
+	private static final float YAW_TARGET_VIBRATION_TOLERANCE = 3; // degrees
+	private final Pose2d SPEAKER_POSE = new Pose2d(0.0, 5.55, Rotation2d.fromRotations(0));
+
+	private DrivebaseSubsystem drivebaseSubsystem;
+	private LauncherSubsystem launcherSubsystem;
+	private IntakeSubsystem intakeSubsystem;
+	private Command yawAlignmentCommand;
+	private Rotation2d yawTarget;
+	private BooleanSupplier launch;
+	private Controls controls;
+
+	public FullTargetCommand(
+			LauncherSubsystem launcherSubsystem,
+			IntakeSubsystem intakeSubsystem,
+			DrivebaseSubsystem drivebaseSubsystem,
+			BooleanSupplier launch) {
+		this.launcherSubsystem = launcherSubsystem;
+		this.intakeSubsystem = intakeSubsystem;
+		this.drivebaseSubsystem = drivebaseSubsystem;
+		this.launch = launch;
+		yawAlignmentCommand = drivebaseSubsystem.rotateToAngle(() -> yawTarget, false);
+
+		addRequirements(launcherSubsystem, intakeSubsystem);
+	}
+
+	@Override
+	public void initialize() {
+		CommandScheduler.getInstance().schedule(yawAlignmentCommand);
+		intakeSubsystem.feederStop();
+		Robot robot = Robot.getInstance();
+		controls = robot.controls;
```

Comment:
Would it be cleaner to pass the `Controls` instance in the constructor (or even just pass a `DoubleSupplier` with the rumble setter)?